### DISCUSSION TITLE
Nextflow

### DISCUSSION THREAD

**Discussion by AlexVCaron** [neurodesk/neurodesk.github.io#84] [General]
Are there any plans to include [Nextlow](nextflow.io) and its companion suite ([nf-core](nf-co.re), [nf-test](nf-test.com)) to Neurodesk ? I'm putting in place a workflow development ecosystem for neuroimaging ([nf-neuro](github.com/scilus/nf-neuro)) and it would be great to be able to use Neurodesk for deployment and to share our work 🙂

I would be happy to help by the way !

**Comment by stebo85** [comment]
Yes - that's possible. Can you make a pull request against the neurodesktop docker image to integrate these tools?

**Comment by stebo85** [comment]
Dear @AlexVCaron 

Any progress on this? Can you describe what needs to be added to make this work and how we can test if it worked?

The additions would need to be made to this docker file: https://github.com/NeuroDesk/neurodesktop/blob/main/Dockerfile

**Comment by stebo85** [comment]
https://github.com/NeuroDesk/neurodesktop/issues/40

**Comment by stebo85** [comment]
Dear @AlexVCaron - we added support for Nextflow in Neurodesk. Can you test it in the latest image build? 
```
docker volume create neurodesk-home &&
sudo docker run \
  --shm-size=1gb -it --security-opt apparmor=neurodeskapp --privileged --user=root --name neurodesktop \
  -v ~/neurodesktop-storage:/neurodesktop-storage \
  --mount source=neurodesk-home,target=/home/jovyan \
  -e NB_UID="$(id -u)" -e NB_GID="$(id -g)" \
  -p 8888:8888 \
  -e NEURODESKTOP_VERSION=2026-02-13 vnmd/neurodesktop:2026-02-13
```

or through: https://play-america.neurodesk.org/

**Reply by stebo85** [reply]
@AlexVCaron - did you get a chance to test?

**Reply by AlexVCaron** [reply]
Sorry I didn't get back to you in time, fully booked right now ! I'll try to take some time to test it before weekend

**Reply by stebo85** [reply]
and would this be a good example for using Nextflow? https://github.com/neurodesk/neurodeskedu/blob/main/books/examples/workflows/nextflow_neurodesk.ipynb
